lib/demos/material/dialog_demo.dart

```diff
@@ -102,6 +102,7 @@ class _DialogDemoState extends State<DialogDemo> with RestorationMixin {
         .copyWith(color: theme.textTheme.caption!.color);
 
     return DialogRoute<String>(
+      anchorPoint: Offset.infinite,
```

Comment:
I did a bit more than that and added a new configuration to the dialog demo, that makes it clear how to choose the screen that is used for displaying the dialog. Please have a look since it is unreviewed code.